services/usb-livesync-service-base.ts

```diff
@@ -155,45 +131,61 @@ export class UsbLiveSyncServiceBase implements IUsbLiveSyncServiceBase {
 		}).future<void>()();
 	}
 
-	private syncCore(platform: string, projectFiles: string[], appIdentifier: string, projectFilesPath: string,
-		platformSpecificLiveSyncServices: IDictionary<any>,
-		notInstalledAppOnDeviceAction: (_device1: Mobile.IDevice) => IFuture<boolean>,
-		beforeLiveSyncAction?: (_device2: Mobile.IDevice, _deviceAppData1: Mobile.IDeviceAppData) => IFuture<void>,
-		shouldRestartApplication?: (_localToDevicePaths: Mobile.ILocalToDevicePathData[]) => IFuture<boolean>): IFuture<void> {
+	protected shouldSynciOSSimulator(platform: string): IFuture<boolean> {
 		return (() => {
-			platform = platform ? this.$mobileHelper.normalizePlatformName(platform) : this.$devicesService.platform;
-			let deviceAppData = this.$deviceAppDataFactory.create(appIdentifier, platform);
-			let localToDevicePaths = _(projectFiles)
-				.map(projectFile => this.getProjectFileInfo(projectFile))
-				.filter(projectFileInfo => projectFileInfo.shouldIncludeFile)
-				.map(projectFileInfo => this.$localToDevicePathDataFactory.create(projectFileInfo.fileName, projectFilesPath, projectFileInfo.onDeviceName, deviceAppData.deviceProjectRootPath))
-				.value();
+			return this.$hostInfo.isDarwin && platform.toLowerCase() === "ios" && (this.$options.emulator || this.$iOSEmulatorServices.isSimulatorRunning().wait());
+		}).future<boolean>()();
+	}
+
+	protected createLocalToDevicePaths(platform: string, appIdentifier: string, projectFilesPath: string, projectFiles: string[]): Mobile.ILocalToDevicePathData[] {
+		let deviceAppData =  this.$deviceAppDataFactory.create(appIdentifier, this.$mobileHelper.normalizePlatformName(platform));
+		let localToDevicePaths = _(projectFiles)
+			.map(projectFile => this.getProjectFileInfo(projectFile))
+			.filter(projectFileInfo => projectFileInfo.shouldIncludeFile)
+			.map(projectFileInfo => this.$localToDevicePathDataFactory.create(projectFileInfo.fileName, projectFilesPath, projectFileInfo.onDeviceName, deviceAppData.deviceProjectRootPath))
+			.value();
+
+		return localToDevicePaths;
+	}
+
+	protected transferFiles(device: Mobile.IDevice, deviceAppData: Mobile.IDeviceAppData, localToDevicePaths: Mobile.ILocalToDevicePathData[]): IFuture<void> {
+		return (() => {
+			this.$logger.info("Transferring project files...");
+			device.fileSystem.transferFiles(deviceAppData.appIdentifier, localToDevicePaths).wait();
+			this.$logger.info("Successfully transferred all project files.");
+		}).future<void>()();
+	}
+
+	private syncCore(data: ILiveSyncData, projectFiles: string[]): IFuture<void> {
+		return (() => {
+			let platform = data.platform ? this.$mobileHelper.normalizePlatformName(data.platform) : this.$devicesService.platform;
+			let deviceAppData =  this.$deviceAppDataFactory.create(data.appIdentifier, this.$mobileHelper.normalizePlatformName(data.platform));
```

Comment:
On the previous line, platform was set to `this.$mobileHelper.normalizePlatformName(data.platform)` and therefore can be used here for shorter code.